message_subscribe_email/message_subscribe_email.module

```diff
@@ -55,6 +55,14 @@ function message_subscribe_email_message_subscribe_get_subscribers_alter(array &
     $flag_ids[] = $flag->id();
   }
 
+  // If email is one of the default notifiers, it should be unset prior to
```

Comment:
The default notifier is added by `message_subscribe` itself, over in `Subscribers::getSubscribers()`, immediately before calling the alter hook.

I'm going to rework this looping to make it clearer what's going on.